lstchain/image/muon/intensity_fitter.py

```diff
@@ -0,0 +1,562 @@
+    return mirror_length - hole_length
+
+
+def pixels_on_ring(radius, pixel_diameter):
+    """Calculate number of pixels of diameter ``pixel_diameter`` on the circumference
+    of a circle with radius ``radius``
+    """
+    circumference = 2 * np.pi * radius
+    n_pixels = u.Quantity(circumference / pixel_diameter)
+    return int(n_pixels.to_value(u.dimensionless_unscaled))
+
+
+@lru_cache(maxsize=1000)
+def linspace_two_pi(n_points):
+    return np.linspace(-np.pi, np.pi, n_points)
+
+
+def create_profile(
+    mirror_radius,
+    hole_radius,
+    impact_parameter,
+    radius,
+    phi,
+    pixel_diameter,
+    oversampling=3,
+):
+    """
+    Perform intersection over all angles and return length
+
+    Parameters
+    ----------
+    impact_parameter: float
+        Impact distance from mirror center
+    ang: ndarray
+        Angles over which to integrate
+    phi: float
+        Rotation angle of muon image
+
+    Returns
+    -------
+    ndarray
+        Chord length for each angle
+    """
+    circumference = 2 * np.pi * radius
+    pixels_on_circle = int(circumference / pixel_diameter)
+
+    ang = phi + linspace_two_pi(pixels_on_circle * oversampling)
+
+    length = intersect_circle(mirror_radius, impact_parameter, ang, hole_radius)
+    length = correlate1d(length, np.ones(oversampling), mode="wrap", axis=0)
+    length /= oversampling
+
+    return ang, length
+
+
+def image_prediction(
+    mirror_radius,
+    hole_radius,
+    impact_parameter,
+    phi,
+    center_x,
+    center_y,
+    radius,
+    ring_width,
+    pixel_x,
+    pixel_y,
+    pixel_diameter,
+    oversampling=3,
+    min_lambda=300 * u.nm,
+    max_lambda=600 * u.nm,
+):
+    """
+    Parameters
+    ----------
+    impact_parameter: quantity[length]
+        Impact distance of muon
+    center_x: quantity[angle]
+        Muon ring center in telescope frame
+    center_y: quantity[angle]
+        Muon ring center in telescope frame
+    radius: quantity[angle]
+        Radius of muon ring in telescope frame
+    ring_width: quantity[angle]
+        Gaussian width of muon ring
+    pixel_x: quantity[angle]
+        Pixel x coordinate in telescope
+    pixel_y: quantity[angle]
+        Pixel y coordinate in telescope
+
+    Returns
+    -------
+    ndarray:
+        Predicted signal
+    """
+    return image_prediction_no_units(
+        mirror_radius.to_value(u.m),
+        hole_radius.to_value(u.m),
+        impact_parameter.to_value(u.m),
+        phi.to_value(u.rad),
+        center_x.to_value(u.rad),
+        center_y.to_value(u.rad),
+        radius.to_value(u.rad),
+        ring_width.to_value(u.rad),
+        pixel_x.to_value(u.rad),
+        pixel_y.to_value(u.rad),
+        pixel_diameter.to_value(u.rad),
+        oversampling=oversampling,
+        min_lambda_m=min_lambda.to_value(u.m),
+        max_lambda_m=max_lambda.to_value(u.m),
+    )
+
+
+def image_prediction_no_units(
```

Comment:
what was the motivation for not using astropy units right away? perf?

Reply:
Yes, this function is called repeatedly by the minuit minimizer and astropy units were a dominant factor

Reply:
Thanks.